packages/autorest.python/autorest/codegen/serializers/parameter_serializer.py

```diff
@@ -96,14 +96,65 @@ def serialize_path(
             [
                 '    "{}": {},'.format(
                     path_parameter.wire_name,
-                    self.serialize_parameter(path_parameter, serializer_name),
+                    ParameterSerializer.serialize_parameter(
+                        path_parameter, serializer_name
+                    ),
                 )
                 for path_parameter in parameters
             ]
         )
         retval.append("}")
         return retval
 
+    @staticmethod
+    def serialize_query_header(
+        param: Parameter, kwarg_name: str, serializer_name: str
+    ) -> List[str]:
+        if (
+            param.location == ParameterLocation.HEADER
+            and param.wire_name.lower() == "repeatability-request-id"
+        ):
+            return [
+                """if "Repeatability-Request-ID" not in _headers:""",
+                """    _headers["Repeatability-Request-ID"] = str(uuid.uuid4())""",
+            ]
+        if (
+            param.location == ParameterLocation.HEADER
+            and param.wire_name.lower() == "repeatability-first-sent"
+        ):
+            return [
+                """if "Repeatability-First-Sent" not in _headers:""",
+                """    _headers["Repeatability-First-Sent"] = _SERIALIZER.serialize_data(datetime.datetime.now(),
+                "rfc-1123")""",
+            ]
+        if param.location == ParameterLocation.HEADER and (
+            param.wire_name.lower() == "client-request-id"
+            or param.wire_name.lower() == "x-ms-client-request-id"
+        ):
+            return []
+        if (
+            param.location == ParameterLocation.HEADER
+            and param.wire_name.lower() == "return-client-request-id"
+        ):
+            return [
+                """if "return-client-request-id" not in _headers:""",
+                """    _headers["return-client-request-id"] = _SERIALIZER.serialize_data(True, "bool")""",
```

Comment:
We should ignore return-client-request-id. People that needs this behvaior will use custom headers:
```python
client.do_something(param, headers={"return-client-request-id": False})
```

If we get feedback that people are angry at this, we can add this back